python/ray/experimental/channel/intra_process_channel.py

```diff
@@ -13,24 +12,23 @@ class IntraProcessChannel(ChannelInterface):
     IntraProcessChannel is a channel for communication between two tasks in the same
     worker process. It writes data directly to the worker's _SerializationContext
     and reads data from the _SerializationContext to avoid the serialization
-    overhead and the need for reading/writing from shared memory.
+    overhead and the need for reading/writing from shared memory. Note that if the
+    readers may mutate the data, users should deep copy the data themselves to avoid
+    side effects.
 
     Args:
-        actor_handle: The actor handle of the worker process.
+        num_readers: The number of readers that will read from this channel.
     """
 
     def __init__(
         self,
-        actor_handle: ray.actor.ActorHandle,
+        num_readers: int = 0,
```

Comment:
Should this be default to 1? What is the use case for 0?

Reply:
There is no use case for 0. I set it to 0 just to ensure that if users forget to set it, the `set_data` will fail. Maybe I can make it a positional argument instead to force users to specify it explicitly.

Reply:
I changed `num_readers` from a keyword argument to a positional argument. https://github.com/ray-project/ray/pull/46431/commits/ff34fd84e7f473d4c5e3b24e677da8f041ffc078